react/react2/react2-week1/app/epic/page.js

```diff
@@ -0,0 +1,61 @@
+"use client";
+import { useSearchParams } from "next/navigation";
+import { useEffect, useState } from "react";
+
+export default function Epic() {
+  const searchParams = useSearchParams();
+  const date = searchParams.get("date");
+
+  const [data, setData] = useState([]);
+
+  useEffect(() => {
+    fetch(
+      "https://api.nasa.gov/EPIC/api/natural/images?api_key=bjGJ41igTdD6HiSXcGVAaTGQ4TpvKhkl54NcWjpU"
```

Comment:
There is actually an API Route that allows you to fetch images from the exact date you want. In that case all hard work with filtering is on their side, and you won't load client so much (also a performance tip, but also a good practice: request as less data a possible)

https://api.nasa.gov/#EPIC